inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -329,7 +400,8 @@ void MultiDeviceExecutableNetwork::run(Task inferPipelineTask) {
 
 MultiDeviceExecutableNetwork::~MultiDeviceExecutableNetwork() {
     // this is necessary to guarantee member destroyed after getting future
-    if (_workModeIsAUTO) {
+    if (_workModeIsAUTO && _loadContext[CPU].isEnabled) {
+        _loadContext[CPU].future.get();
```

Comment:
Yes, we didn't use future.valid(), so it's same to use wait and get here, 
about exception in load network has be handled, it will not throw exception again.
A remind, the loading process contain "loadNetwork" and "GenerateWorkers"， if "GenerateWorkers" thrown exception, it also will cause the app exit